### PR TITLE
Changed default log.sqlite3 directory to be more semantic.

### PR DESCRIPTION
**Pull request by undercase**
The default path for the sqlite 3 log file has been changed to be in the root directory of the user's application, when none is provided.

### CODE REVIEW

**Comment by GreenVars** [comment]
This is better suited in the script file than in DiaryDB. It's a slippery slope using command line arguments outside of the bin directory because the code becomes less maintainable.

**Comment by undercase** [comment]
This doesn't use command line arguments. This has nothing to do with the command line - before it assumed the db was at 'log.sqlite3', now it has a smarter algorithm that checks the root directory of the project.

**Comment by GreenVars** [comment]
What is the use of sys.argv[0] here?

**Comment by undercase** [comment]
sys.argv[0] retrieves the name of the root (or __main__) python file.

**Comment by GreenVars** [comment]
when would sys.arv[0] result to false? Is there really a need for this tweak, I feel like it may be a step towards background control that is heinous.

**Comment by undercase** [comment]
There are some corner cases where a python file is invoked without a filename (these are really rare). In that case sys.argv[0] would evaluate to empty or false, and the old path being used as the default ('log.sqlite3') would be used as the default.

This will make it so that if they want, they can just run the generate command or use DiaryDB with no setup of file paths at all. They can still tweak it, as if a specific path is provided that is always used.

**Comment by GreenVars** [comment]
If this is specifically for the generate command, why should it be in DiaryDB and not the bin/diary file? What benefit does this have outside the generate command use?

**Comment by undercase** [comment]
It's not for the generate command, it's so that people can use the default name ('log.sqlite3') in the base of the application and have it automatically find their file for them. It is also programmatically impossible to put this in the command file, as this can only be invoked when they actually create the DiaryDB instance in their application code.

Regardless this isn't tied to the generate command, it is useful on a general scale as a default.

**Comment by GreenVars** [comment]
So be it. Some tests are needed however.

**Comment by undercase** [comment]
A lot of developers prefer to have a no-setup or almost no-setup system - if they can invoke DiaryDB without passing a filename and it has useful defaults they will be more likely to use the framework. Look at frameworks like rails or flask that have similar default systems set up.

**Comment by undercase** [comment]
There are already tests covering finding the file - when I added the static 'log.sqlite3' default I made sure of that.

File: diary/logdb.py [hunk 1]
```diff
@@ -10,12 +11,18 @@ class DiaryDB(object):
     to log. DiaryDB uses SQLite3.
     """
 
-    def __init__(self, path='log.sqlite3'):
+    def __init__(self, path=None):
         """
         Create the connection with the database and attempt to make a table.
         :param path: relative path of database
```
Comment (GreenVars) [inline diary/logdb.py]: These docs should probably get a small update to reflect this change.

**Comment by GreenVars** [comment]
Coveralls disagrees. https://coveralls.io/builds/7634384/source?filename=diary%2Flogdb.py#L20

**Comment by undercase** [comment]
Coverall does a really shallow analysis - in the command tests the default is used, so it functionally tests the new code. I've just added a commit that reflects that change.

File: diary/logdb.py (hunk 1 above)
Comment (undercase) [inline diary/logdb.py]: Just added documentation update.

**Comment by GreenVars** [comment]
It can't hurt to add a couple more tests which give non-default arguments. Code can't be committed without testing reported

**Comment by undercase** [comment]
@GreenVars I've added more testing and fixed some errors - it should be good to go.

File: diary/logdb.py [hunk 2]
```diff
@@ -69,6 +69,10 @@ def close(self):
         except sqlite3.ProgrammingError:
             pass
 
+    def destroy(self):
```
Comment (GreenVars) [inline diary/logdb.py]: We don't really have to have this here and probably shouldnt

Comment (undercase) [inline diary/logdb.py]: On second thought, I agree - it's been removed.

Comment (undercase) [inline diary/logdb.py]: @GreenVars Any other concerns?

Comment (GreenVars) [inline diary/logdb.py]: Mostly the failing build.

Comment (undercase) [inline diary/logdb.py]: The build works on my computer, I'll look into the issue Travis CI is having.

I'm getting the feeling that you don't really want me contributing to the project. You seemed excited about open source, as am I, but you've been very hostile to my changes and haven't been responsive to my requests for comments / concerns. Please tell me if I am misguided in this assessment.

Comment (GreenVars) [inline diary/logdb.py]: Very misguided. I'm incredibly happy to see you contributing and I like the changes you have made. This is my first open source project and I'm trying to prioritize some things. Regardless, I'll try to be more responsive and assist you.

Comment (undercase) [inline diary/logdb.py]: That's really great to hear :)! Most of your messages were very terse, and I was worried that you didn't want me to contribute. Thanks for reassuring me.

**Comment by GreenVars** [comment]
Made more tests and fixed the travis failure.

**Comment by GreenVars** [comment]
Merged.

**Pull request merged**